### ISSUE TITLE
Unexpected behavior from classdiagram class fields containing parentheses inside of brackets

### ISSUE THREAD

**Issue by diceroll123** [mermaid-js/mermaid#5881]
### Description

It seems that after parentheses close, anything after that is strictly put after the `:` for the class field.

I got this from a valid Python type annotation, https://github.com/fastapi/fastapi/blob/7c6f2f8fde68f488163376c9e92a59d46c491298/fastapi/datastructures.py#L64-L73 (see third line in my repro)

### Steps to reproduce

```mermaid
classDiagram
    class Thing {
        + a()] blah
        + Literal[a("blah!")] blah
        + Annotated[Optional[str], Doc("The original file name.")] filename
    }
```

### Setup

- Mermaid version:
- Browser and Version: [Chrome, Edge, Firefox]

**Comment by jgreywolf** [comment]
Since I am not sure if I am fully understanding, can you please explain what you expected to happen, vs what did happen?  Also, if you could provide the exact syntax you used it will be very helpful for repro

**Comment by diceroll123** [comment]
Sure thing, my bad!

I've updated the original post with unrendered markdown of the diagram.

Given that class diagram, I expected something like the following image, which I've edited from the initial render with Inspect Element,

<img width="525" alt="image" src="https://github.com/user-attachments/assets/1380364e-2e55-4d9f-930e-1e4fc2046ab8">

---

_additionally, however_

I am only now realizing that by these annotated types having a complete pair of parens, that they are marked as functions/methods (third compartment) rather than attributes (middle compartment).

While this `Doc` method (the third line in the rendered mermaid diagram) within type annotations is not (yet?) an approved part of Python, it is backward-compatible + valid syntax and is already in use, https://peps.python.org/pep-0727/

---

To summarize:

Syntactically correct Python class annotations which themselves execute a callable function _such as `filename` below_, due to having a pair of parens are both:
- classified as a callable itself
- incorrectly parsed once the parens end (as shown with the bracket on the right-hand side of the colon between the annotation and the variable name)

within a rendered mermaid diagram.

```python
from typing import Optional, Annotated
from typing_extensions import Doc

class Thing:
    filename: Annotated[Optional[str], Doc("The original file name.")]
```

**Comment by jgreywolf** [comment]
I am sorry for the LONG delay - but I am now actively working on an update to the class diagram syntax that will address this, and other, issues.

Should have something within a couple of days :)

**Comment by jgreywolf** [comment]
Its been more than a couple of days.  I am going to split this out from the large list of class diagram issues I was trying to put into one PR ;)
